initrd/bin/network-init-recovery

```diff
@@ -4,58 +4,104 @@
 
 TRACE_FUNC
 
-# bring up the ethernet; maybe should do DHCP?
+# bring up the ethernet interface
 ifconfig lo 127.0.0.1
 
+echo "Loading Ethernet network modules..."
 network_modules="e1000 e1000e igb sfc mdio mlx4_core mlx4_en"
-for module in `echo $network_modules`; do
+for module in $(echo $network_modules); do
 	if [ -f /lib/modules/$module.ko ]; then
-		insmod /lib/modules/$module.ko
+		#check if module is already loaded and load it if not
+		lsmod | grep -q $module || insmod /lib/modules/$module.ko
 	fi
 done
 
-if [ -e /sys/class/net/eth0 ]; then
+#Tethering over USB for Mobile phones supporting CDC (Android Pixel 6a+, Librem phone, etc.)
+if [ -e /lib/modules/cdc_ether.ko ]; then
+	#first enable USB controllers
+	enable_usb
+	echo ""
+	echo "Please verify that your mobile (CDC NCM/EEM tethering compatible phone) is networked in the desired way (WIFI/mobile + VPN/Orbot/etc)"
+	echo "Please connect mobile phone to this machine's fast USB port (blue identified) through a known working data cable"
+	echo "Please enable USB tethering prior of going further (Android: select 'Charging this device via USB' notification and enable tethering option)"
+	read -p "Press Enter to continue now or wait 30 seconds..." -n 1 -r -t 30
+	
+	network_modules="mii usbnet cdc_ether cdc_ncm cdc_eem"
+	echo "Loading USB tethering network related modules: $network_modules..."
+	for module in $(echo $network_modules); do
+		if [ -f /lib/modules/$module.ko ]; then
+			#check if module is already loaded and load it if not
+			lsmod | grep -q $module || insmod /lib/modules/$module.ko
+		fi
+	done
+	if ! [ -e /sys/class/net/usb0 ]; then
+		echo ""
+		echo "Tethering USB network interface was NOT detected with loaded kernel modules : $network_modules"
+		echo "Please check your phone's linux drivers requirements"
+		echo "Note that RNDIS kernel module inclusion was discussed and rejected due to security implications and planned deprecation under Linux kernel altogether"
+		echo "CDC NCM/CDC EEM support is known to be available on a majority of Android/GrapheneOS as well as Librem phones"
+		echo "Non-exhaustive exeptions: Pixel 4a* known to only tether over RNDIS and won't be supported"
+		echo "Apple phones won't be supported due to size and complexity of the drivers and toolstack required to support tethering"
+		read -p "Press Enter to continue now or wait 30 seconds..." -n 1 -r -t 30
```

Comment:
Same here, I think we should change this to just Press Enter, take out the timeout.